src/components/tables/components/row-create-username/row-create-username.tsx

```diff
@@ -1,69 +1,63 @@
 import { useState, type FC, useEffect } from "react";
-import {
-  Alert,
-  Avatar,
-  Button,
-  Stack,
-  TableCell,
-  TableRow,
-  TextField,
-  Typography,
-} from "@mui/material";
+import { Stack } from "@mui/material";
+import { z } from "zod";
 
-import { text } from "~/assets/text/index.js";
 import { getRandomUsername } from "~/utils/username.js";
+import { validUsername, validation } from "~/constants/validation.js";
+import { profanityFilter } from "~/service/index.js";
+import { UsernameInputField } from "~/components/input-field/input-field.jsx";
 
 import { styles } from "./styles.js";
 
 interface RowCreateUsernameProps {
-  onSetUsername: (username: string) => Promise<void>;
-  error?: string;
+  submitUsername: (username: string) => Promise<void>;
 }
 
 export const RowCreateUsername: FC<RowCreateUsernameProps> = ({
-  onSetUsername,
-  error,
+  submitUsername,
 }) => {
   const [username, setUsername] = useState("");
+  const [error, setError] = useState("");
 
   useEffect(() => {
     const name = getRandomUsername();
     setUsername(name.replace(/[0-9]/g, ""));
   }, []);
 
+  const validateUsername = (name: string) => {
```

Comment:
Apparently you can do something like this to simplify the logic:
```TS 
const usernameSchema = z.string().min(validation.username.min, validation.username.error.tooShort)
                             .max(validation.username.max, validation.username.error.tooLong)
                             .regex(/^[a-zA-Z0-9_\-.]*$/, validation.username.error.specialCharacters);

const validateUsername = (name) => {
  let error = "";

  // Combine regex, min, max checks into one
  const result = usernameSchema.safeParse(name);
  if (!result.success) {
    error = result.error.issues[0].message; // Get the first validation error message
  } else if (profanityFilter.exists(name)) {
    error = validation.username.error.profanity;
  } else if (name.includes(" ")) {
    error = validation.username.error.space;
  }

  setError(error); // Assuming setError is defined elsewhere to handle setting the error message
};
```

Reply:
oh, sweet I'll use that